From: Izik Eidus <ieidus@redhat.com>
Date: Thu, 8 Oct 2009 14:39:39 +0000 (+0200)
Subject: ksm support
X-Git-Url: http://git.kernel.org/?p=virt%2Fkvm%2Fqemu-kvm.git;a=commitdiff_plain;h=ccb167e9d7d460a7cd09fdabd848efd3606dd27e

ksm support

patch is slighly modified by <support@proxmox.com> for old libc 

Call MADV_MERGEABLE on guest memory allocations.  MADV_MERGABLE will be
available starting in Linux 2.6.32.  This system call registers a region of
virtual address space with Linux as a candidate for transparent memory
sharing.

Patchworks-ID: 35447
Signed-off-by: Izik Eidus <ieidus@redhat.com>
Signed-off-by: Anthony Liguori <aliguori@us.ibm.com>
---

Index: qemu-kvm/exec.c
===================================================================
--- qemu-kvm.orig/exec.c	2009-12-21 10:37:57.000000000 +0100
+++ qemu-kvm/exec.c	2009-12-21 11:22:34.000000000 +0100
@@ -48,6 +48,9 @@
 #include <qemu.h>
 #endif
 
+/* our libc does not define it */
+#define MADV_MERGEABLE   12
+
 //#define DEBUG_TB_INVALIDATE
 //#define DEBUG_FLUSH
 //#define DEBUG_TLB
